src/pathpyG/core/Graph.py

```diff
@@ -116,11 +145,16 @@ def to_undirected(self) -> None:
             ```py
             import pathpyG as pp
             g = pp.Graph(torch.LongTensor([[1, 1, 2], [0, 2, 1]]))
-            g.to_undirected()
+            g_u = g.to_undirected()
             ```
         """
         tf = ToUndirected()
-        self.data = tf(self.data)
+        return Graph(tf(self.data), self.mapping)
+
+    def to_weighted_graph(self) -> Graph:
```

Comment:
This method suggests that we allow multiedges in our graph. But then `self.edge_to_index` (line 76) will be ambiguos.